dev/docker/ballista-builder.Dockerfile

```diff
@@ -15,7 +15,7 @@
 # specific language governing permissions and limitations
 # under the License.
 
-FROM rust:1.74.1-buster
+FROM rust:1.81-bullseye
```

Comment:
`buster`'s package repo did not have a high enough version of `protoc` that supports `experimental_allow_proto3_optional` flag, that's why I bumped